mailet/api/src/main/java/org/apache/mailet/AttributeValue.java

```diff
@@ -101,12 +106,14 @@
     @SuppressWarnings({ "unchecked", "rawtypes" })
     public static AttributeValue<Collection<AttributeValue<?>>> of(Collection<AttributeValue<?>> value) {
         Preconditions.checkNotNull(value, "value should not be null");
+        Preconditions.checkArgument(value.stream().allMatch(x -> x instanceof AttributeValue), "values should be instances of Attributes");
```

Comment:
I think you dont need this check, It's guaranteed from compiling time. A `value` passed to this `of()` method is always be a collection of `AttributeValue`. Don't you think?

Reply:
I'd agree with you, but Java casting system is broken and in some parts of the code, we check `something instanceof Collection` and we directly cast it to `Collection<Attribute<?>>`, which leads to `CastException` when we use it, in a very different place. It happened numerous of time and it quite painful to debug.

Reply:
even you putted your Preconditions check, it always throws `ClassCastException` directly from `of()`

Reply:
Yes, I'm struggling to find a way to not make it cast here.

Reply:
You want to force this at compile time, anyway, I've casted it to `Collection` and it seems ok.